pandas/_libs/tslib.pyx

```diff
@@ -373,8 +374,24 @@ class Timestamp(_Timestamp):
                           FutureWarning)
             freq = offset
 
+        if tzinfo is not None:
+            if not PyTZInfo_Check(tzinfo):
+                # tzinfo must be a datetime.tzinfo object, GH#17690
+                raise TypeError('tzinfo must be a datetime.tzinfo object, '
+                                'not %s' % type(tzinfo))
+            elif tz is not None:
+                raise ValueError('Can provide at most one of tz, tzinfo')
+            elif ts_input is not _no_input:
+                raise ValueError('When creating a Timestamp with this type '
```

Comment:
misspell

what case is this trying to catch? this is a confusing error message

Reply:
Good catch on the spell-check, thanks.

Under the status quo, the `tzinfo` kwarg is used in only one branch:
```
        if ts_input is _no_input:
            # User passed keyword arguments.
            return Timestamp(datetime(year, month, day, hour or 0,
                                      minute or 0, second or 0,
                                      microsecond or 0, tzinfo),
                             tz=tzinfo)
```

In all other cases `tzinfo` is silently ignored.  This raises instead of ignoring the input.  As you noted below, it may be better to change this behavior.

You're right the message is oddly worded.  "this type of inputs" in this case is "anything other than the keyword arguments for datetime(...)".  Suggestions for wording?  (unnecessary if we change the behavior instead)